1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,17 +1,58 @@
-function setAlarm() {}
+const pauseBtn = document.getElementById("pause");
+const countdownDisplay = document.getElementById("timeRemaining");
+let isPaused = false;
+var audio = new Audio("alarmsound.mp3");
+let flashingBackground;
 
-// DO NOT EDIT BELOW HERE
+function pauseCountdown() {
+  if (isPaused) {
+    isPaused = false;
+    pauseBtn.innerText = "Pause";
+  } else {
+    isPaused = true;
+    pauseBtn.innerText = "Continue";
+  }
+}
 
-var audio = new Audio("alarmsound.mp3");
+function setAlarm() {
+  pauseAlarm();
+  isPaused = false;
+  pauseBtn.innerText = "Pause";
+
+  let time = document.getElementById("alarmSet").value;
+  let counter = setInterval(updateCountdown, 1000);
+
+  function updateCountdown() {
+    let minutes = Math.floor(time / 60);
+    let seconds = time % 60;
+    minutes = minutes < 10 ? "0" + minutes : minutes;
```

Comment:
this is an interesting point, when minutes are less than 10 the variable becomes a string. I have not yet examples where I would get a NaN because numeric conversion happens in mathematical functions and expressions automatically.

Reply:
You're right that in this situation you'll probably not get a NaN, but if you can start writing code defensively against Javascript's automatic [type conversion rules](https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/) which are quite complex, then you'll avoid more bugs in the future.